xarray/core/dataarray.py

```diff
@@ -6288,6 +6288,37 @@ def argmax(
         else:
             return self._replace_maybe_drop_dims(result)
 
+    def argtopk(
```

Comment:
I think `generate_aggregations.py` would make sense here so that we add it everywhere with the same docstring. That would get us groupby  support for example, and I can eventually plug in flox when https://github.com/xarray-contrib/flox/pull/374/files is ready